### PR TITLE
BREAKING: Removing stylistic data-attributes

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

These were a tech debt from a while ago cleaning it up.

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/afraid-pugs-kiss.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@aws-amplify/ui-react": major
+---
+
+BREAKING: 
+* Removing stylistic data-attributes from React primitives
+* Update SwitchField classnames to be BEM
+* Removing remenants of "countryCode"
```
Comment (reesscot) [inline .changeset/afraid-pugs-kiss.md:8]:
```suggestion
* Removing remnants of "countryCode"
```

File: packages/react/src/components/AccountSettings/ChangePassword/__tests__/__snapshots__/ChangePassword.test.tsx.snap [hunk 2]
```diff
@@ -43,7 +41,6 @@ exports[`ChangePassword renders as expected 1`] = `
               aria-checked="false"
               aria-label="Show password"
               class="amplify-button amplify-field-group__control amplify-field__show-password"
-              data-fullwidth="false"
```
Comment (reesscot) [inline packages/react/src/components/AccountSettings/ChangePassword/__tests__/__snapshots__/ChangePassword.test.tsx.snap:46]: This isn't used either?

**Review by reesscot — APPROVED** [review]
Approved, assuming my comment above is not relevant

**Review by calebpollman — COMMENTED** [review]
LGTM! Left a couple clarifying questions about `data` attributes that are remaining, and saw that some unit test names reference the `data` attributes that no longer exist (would be good to audit the related unit test names in general, didn't comment on all of them)

File: packages/react/src/components/AccountSettings/ChangePassword/__tests__/__snapshots__/ChangePassword.test.tsx.snap (hunk 2 above)
Comment (dbanksdesign) [reply packages/react/src/components/AccountSettings/ChangePassword/__tests__/__snapshots__/ChangePassword.test.tsx.snap:46]: Correct, doing a search in the repo for `data-fullwidth` only comes up with snapshot tests and vue/angular code. No CSS references it.

File: packages/react/src/primitives/Divider/Divider.tsx [hunk 3]
```diff
@@ -28,7 +28,6 @@ const DividerPrimitive: Primitive<DividerProps, 'hr'> = (
       aria-orientation={orientation}
       as="hr"
       className={componentClasses}
-      data-size={size}
       data-label={label}
```
[the remark this answers is not in the document]
Comment (dbanksdesign) [reply packages/react/src/primitives/Divider/Divider.tsx:31]: `data-label` is actually used in the CSS to render that text: https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/theme/css/component/divider.scss#L40

File: packages/react/src/primitives/SwitchField/SwitchField.tsx [hunk 4]
```diff
@@ -85,7 +85,6 @@ const SwitchFieldPrimitive: Primitive<SwitchFieldProps, 'div'> = (
         classNameModifier(ComponentClassName.SwitchField, size),
         className
       )}
-      data-size={size}
       data-label-position={labelPosition}
```
[the remark this answers is not in the document]
Comment (dbanksdesign) [reply packages/react/src/primitives/SwitchField/SwitchField.tsx]: Yea because I didn't want to deal with it... but then I just did

**Review by calebpollman — DISMISSED** [review]
🚢

**Review by calebpollman — APPROVED** [review]
PR feedback handling 10 out of 10, would review again

**Review by calebpollman — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
Thanks for cleaning this up!

**Pull request merged**